test/integration/targets/ec2_snapshot/tasks/main.yml

```diff
@@ -0,0 +1,250 @@
+      ec2_snapshot:
+        volume_id: '{{ volume_id }}'
+        last_snapshot_min_age: 60
+      register: result
+
+    - name: Get info about snapshots
+      ec2_snapshot_info:
+        filters:
+          "tag:Name": '{{ resource_prefix }}'
+      register: info_result
+
+    - assert:
+        that:
+          - result is not changed
+          - info_result.snapshots| length == 1
+
+    - name: Pause so we can do a last_snapshot_min_age test
+      pause:
+        minutes: 1
+
+#    JR: Check mode not supported
+#    - name: Take snapshot if most recent >1min (True) (check mode)
+#      ec2_snapshot:
+#        volume_id: '{{ volume_id }}'
+#        snapshot_tags:
+#          Name: '{{ resource_prefix }}'
+#        last_snapshot_min_age: 1
+#        check_mode: true
+#      register: result
+#    - assert:
+#        that:
+#          - result is changed
+
+    - name: Take snapshot if most recent >1min (True)
+      ec2_snapshot:
+        volume_id: '{{ volume_id }}'
+        last_snapshot_min_age: 1
+      register: result
+
+    - name: Get info about snapshots
+      ec2_snapshot_info:
+        filters:
+          "tag:Name": '{{ resource_prefix }}'
+      register: info_result
+
+    - assert:
+        that:
+          - result is changed
+          - info_result.snapshots| length == 2
+          - '"{{ result.snapshot_id }}" in "{{ info_result| json_query("snapshots[].snapshot_id") }}"'
+
+#    JR: Check mode not supported
+#    - name: Take snapshot with a tag (check mode)
+#      ec2_snapshot:
+#        volume_id: '{{ volume_id }}'
+#        snapshot_tags:
+#          MyTag: '{{ resource_prefix }}'
+#      register: result
+#    - assert:
+#        that:
+#          - result is changed
+
+    #  Wait at least 15 seconds between concurrent volume snapshots.
+    - name: Prevent SnapshotCreationPerVolumeRateExceeded errors
+      pause:
+        seconds: 15
+
+    - name: Take snapshot and tag it
+      ec2_snapshot:
+        volume_id: '{{ volume_id }}'
+        snapshot_tags:
+          MyTag: '{{ resource_prefix }}'
+      register: result
+
+    - name: Get info about snapshots by tag
+      ec2_snapshot_info:
+        filters:
+          "tag:MyTag": '{{ resource_prefix }}'
+      register: tag_info_result
+
+    - assert:
+        that:
+          - result is changed
+          - result.tags| length == 2
+          - tag_info_result.snapshots| length == 1
+          - result.tags.MyTag == "{{ resource_prefix }}"
+          - '"{{ result.snapshot_id }}" in "{{ tag_info_result| json_query("snapshots[].snapshot_id") }}"'
+
+    - name: Get info about all snapshots for this test
+      ec2_snapshot_info:
+        filters:
+          "tag:Name": '{{ resource_prefix }}'
+      register: info_result
+
+    - assert:
+        that:
+          - info_result.snapshots| length == 3
+
+    - name: Delete the tagged snapshot
+      ec2_snapshot:
+        state: absent
+        snapshot_id: '{{ tag_info_result.snapshots[0].snapshot_id }}'
```

Comment:
It feels like pulling this from tag_info_result results in an ugly long lookup here.

Is it worth dumping result.snapshot_id into a fact back when the snapshot is created?  Might be easier to follow what's being deleted.